CriticalMass/TwitterManager.swift

```diff
@@ -34,15 +34,14 @@ class TwitterManager {
 
     public func loadTweets(_ completion: ResultCallback<[Tweet]>? = nil) {
         networkLayer.get(request: request) { [weak self] result in
```

Comment:
I think this early exit saves us from reading three lines with a nil coalescing operator and also assured that self is always there for the scope off the block

Reply:
> In general I think the PR is not really focused and does s lot of changes in different classes. But to be fair I am struggling with that myself 😄

I just wanted to write some tests and ended up with fixing bugs 🙈. To not further grow the scope of this PR, I address PR comments on Issues that are not caused by this PR. I hope that's okay

Reply:
Yes but if we have an active completion block we should always try to call it otherwise the calling function might wait forever and show an infinite loading animation for example.

Reply:
Good point. I missed that one.
The Manager also handing out view states feels weird though